Cargo.toml

```diff
@@ -18,10 +18,11 @@ cfg-if = { version = "1.0.0", features = [] }
 web-sys = { version = "0.3.66", optional = true }
 
 [lib]
+doctest = false  # if you feel like fixing docs go right ahead...
```

Comment:
so when it's `doctest = true` does it fail?

Reply:
Yes:
```

Some errors have detailed explanations: E0425, E0433.
For more information about an error, try `rustc --explain E0425`.
Couldn't compile the test.

failures:
    src/hotkeys_provider.rs - hotkeys_provider::HotkeysContext (line 27)
    src/hotkeys_provider.rs - hotkeys_provider::HotkeysProvider (line 72)
    src/hotkeys_provider.rs - hotkeys_provider::HotkeysProviderProps (line 103)
    src/hotkeys_provider.rs - hotkeys_provider::__HotkeysProvider (line 72)
    src/lib.rs - (line 116)
    src/lib.rs - (line 161)
    src/lib.rs - (line 183)
    src/lib.rs - (line 198)
    src/lib.rs - (line 207)
    src/lib.rs - (line 216)
    src/lib.rs - (line 236)
    src/lib.rs - (line 288)
    src/lib.rs - (line 316)
    src/lib.rs - (line 372)
    src/lib.rs - (line 44)
    src/lib.rs - (line 80)
    src/use_hotkeys.rs - use_hotkeys::parse_key (line 26)
    src/use_hotkeys.rs - use_hotkeys::use_hotkeys_scoped (line 120)

test result: FAILED. 0 passed; 18 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.29s
```

Reply:
I think you can tell rustdoc to ignore code blocks by changing '''rust to '''ignore, but then it will mess with the README.md highlighting. Also some of the links break it.

Reply:
Damn dafuq. 

Oh, it might have to do with the `cfg_if`.

So I had to bring in `cfg_if` to support ssr apps. Leptos apps using server side rendering is a bit weird, where it's either in the `server` state or a `hydrate` state. You can think of the hydrate state as the client side, which affords us the libraries like `web_sys`. When we're on the state side, we don't have the ability to use these libraries.

We may have to edit doctests to run on `hydrate` or `csr` mode.

Reply:
This happened before `cfg_if` was add too. And also the error I posted was when running `cargo test --features "csr"` with the doctest set to true. So yeah, no idea. Everything displays fine though when running `cargo doc --features "csr" --open`

Reply:
I'm just gonna resolve for now. Let me know if you get any ideas